logging/src/main/java/uk/gov/pay/logging/GovUkPayAccessJsonLayoutFactory.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.pay.logging;
+
+import ch.qos.logback.access.spi.IAccessEvent;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.core.LayoutBase;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.dropwizard.logging.json.AccessJsonLayoutBaseFactory;
+import io.dropwizard.logging.json.layout.AccessJsonLayout;
+import io.dropwizard.logging.json.layout.TimestampFormatter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * This programmatically configures the JSON logging to be equivalent to:
+ *
+ *       - type: console
+ *         layout:
+ *           type: access-json
+ *           timestampFormat: "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+ *           CUSTOM_FIELD_NAMES:
+ *             timestamp: "@timestamp"
+ *             userAgent: "user_agent"
+ *             requestTime: "response_time"
+ *             uri: "url"
+ *             protocol: "http_version"
+ *             status: "status_code"
+ *             contentLength: "content_length"
+ *             remoteAddress: "remote_address"
+ *           additionalFields:
+ *             "@version": 1
+ *             
+ * More additional fields can be added as required.
+ */
+@JsonTypeName("govuk-pay-access-json")
+public class GovUkPayAccessJsonLayoutFactory extends AccessJsonLayoutBaseFactory {
+
+    private static final Map<String, String> CUSTOM_FIELD_NAMES = Map.of(
+            "timestamp", "@timestamp",
+            "userAgent", "user_agent",
+            "requestTime", "response_time",
+            "uri", "url",
+            "protocol", "http_version",
+            "status", "status_code",
+            "contentLength", "content_length",
+            "remoteAddress", "remote_address"
+    );
+
+    private static final String TIMESTAMP_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+    @Override
+    public LayoutBase<IAccessEvent> build(LoggerContext context, TimeZone timeZone) {
+        var additionalFields = new HashMap<>(this.getAdditionalFields());
+        additionalFields.put("@version", 1);
```

Comment:
Just wondering if it might be beneficial for this to throw an exception if the map returned by `getAdditionalFields()` doesn't include a key named `container` with a value. That would ensure it's always been set in the config and avoid someone mistakenly not setting it?

Reply:
I think this is a good idea, will add.